### PR TITLE
RC2Fixes

### PR DESCRIPTION
**Pull request by brentschmaltz**
Big code cleanup, mainly in test. The main src fix was JwtPayload.Claims.

Consolidating test code, removed dead code.
JwtPayload - JSON parsing fixes.

#342 
#89 
#74

### CODE REVIEW

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 1]
```diff
@@ -368,6 +368,11 @@ public virtual JwtSecurityToken CreateToken(string issuer = null, string audienc
 
         private JwtSecurityToken CreateJwt(string issuer, string audience, IEnumerable<Claim> claims, DateTime? notBefore, DateTime? expires, DateTime? issuedAt, SigningCredentials signingCredentials)
         {
+            return new JwtSecurityToken(CreateEncodedJwt(issuer, audience, claims, notBefore, expires, issuedAt, signingCredentials));
+        }
+
+        public string CreateEncodedJwt(string issuer, string audience, IEnumerable<Claim> claims, DateTime? notBefore, DateTime? expires, DateTime? issuedAt, SigningCredentials signingCredentials)
```
Comment (tushargupta51) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: Why making it public? They can use the wrap the values in "SecurityTokenDescriptor" and call the public method.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 2]
```diff
@@ -302,19 +302,19 @@ public override bool CanReadToken(string tokenString)
         /// Returns a Json Web Token (JWT).
         /// </summary>
         /// <param name="tokenDescriptor"> a <see cref="SecurityTokenDescriptor"/> that contains details of token contents and <see cref="SignatureProvider"/>.
-        public string CreateJwt(SecurityTokenDescriptor tokenDescriptor)
+        public string CreateEncodedJwt(SecurityTokenDescriptor tokenDescriptor)
```
Comment (tushargupta51) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: This is a breaking change. Do we want to make it at this point?

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 1 above)
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: Just an an alternative.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 2 above)
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: Yes, I want to make this now. Before we ship. I want the name of the method to distinguish if an encoded string OR JwtSecurityToken is returned.

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectConfigurationRetrieverTests.cs [hunk 3]
```diff
@@ -73,30 +73,17 @@ public class OpenIdConnectConfigurationRetrieverTests
...
-        [Fact(DisplayName = "OpenIdConnectConfigurationRetrieverTests: FromText")]
+        [Fact]
         public async Task FromText()
         {
-            OpenIdConnectConfiguration configuration;
-            List<string> errors = new List<string>();
-
-            configuration = await GetConfigurationFromMixedAsync(OpenIdConfigData.OpenIdConnectMetadataPingString, expectedException: ExpectedException.NoExceptionExpected);
+            var context = new CompareContext();
+            var configuration = await GetConfigurationFromMixedAsync(OpenIdConfigData.OpenIdConnectMetadataPingString, expectedException: ExpectedException.NoExceptionExpected);
```
Comment (tushargupta51) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectConfigurationRetrieverTests.cs]: Missing IdentityComparer.AreEqual(configuration, OpenIdConfigData.OpenIdConnectMetadataPingString, context);  here.

File: test/Microsoft.IdentityModel.Tokens.Tests/Defines.cs [hunk 4]
```diff
@@ -1,39 +0,0 @@
...
-{
-    public class Defines
-    {
-#if SAML
-        if saml tokens are enables
-#endif
-#if JWT_XML
-        if reading / writing jwt embeded in xml as in ws fed
-#endif
-    }
```
Comment (tushargupta51) [inline test/Microsoft.IdentityModel.Tokens.Tests/Defines.cs]: Wow! What we had this for?

**Pull request merged**